text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -401,34 +406,52 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
-
-                if (sampling_params.repetition_penalty != 1.0f) {
-                    auto repetition_penalty_transform = RepetitionPenaltyTransform(sampling_params.repetition_penalty);
-                    logit_vector = repetition_penalty_transform.apply(logit_vector, sequence_group->get_unique_generated_ids());
-                }
                 std::vector<Sequence::Ptr> running_sequences = sequence_group->get_running_sequences();
-                OPENVINO_ASSERT(running_sequences.size() == 1);
-
-                int64_t sampled_token_id;
                 if (sampling_params.is_greedy_sampling()) {
-                    sampled_token_id = _greedy_sample(logit_vector);
-                }
-                else {  // .is_multinomial()
-                    sampled_token_id = _multinomial_sample(logit_vector, sampling_params.temperature, sampling_params.top_p, sampling_params.top_k);
+                    OPENVINO_ASSERT(num_running_sequences == 1);
                 }
+                auto register_new_token = [&](const LogitWithIdx& sampled_token_id, Sequence::Ptr running_sequence) {
+                    sequence_group->register_generated_token_id(sampled_token_id.second);
+                    running_sequence->append_token(sampled_token_id.second, sampled_token_id.first);
+
+                    if (sampling_params.max_new_tokens == running_sequence->get_generated_len() ||
+                        sampled_token_id.second == sampling_params.eos_token_id && !sampling_params.ignore_eos) {
+                        // drop sequence from scheduler
+                        sampler_output.m_dropped_sequences.push_back(running_sequence->get_id());
+                    }
+                };
+                for (size_t running_sequence_id = 0; running_sequence_id < num_running_sequences; ++running_sequence_id) {
+                    auto logit_vector = _get_logit_vector(sequence_group_logits, running_sequence_id);
 
-                sequence_group->register_generated_token_id(sampled_token_id);
-
-                // in case of greedy search we always have a single parent sequence to sample from
-                running_sequences[0]->append_token(sampled_token_id, sequence_group_logits.data<const float>()[sampled_token_id]);
+                    if (sampling_params.repetition_penalty != 1.0f) {
+                        auto repetition_penalty_transform = RepetitionPenaltyTransform(sampling_params.repetition_penalty);
+                        logit_vector = repetition_penalty_transform.apply(logit_vector, sequence_group->get_unique_generated_ids());
+                    }
 
-                if (sampling_params.max_new_tokens == running_sequences[0]->get_generated_len() ||
-                    sampled_token_id == sampling_params.eos_token_id && !sampling_params.ignore_eos) {
-                    // stop sequence by max_new_tokens or EOS token
-                    running_sequences[0]->set_status(SequenceStatus::FINISHED);
```

Comment:
why has it moved to `sequence_group->notify_handle` ?

Reply:
Discussed offline. TODO: support streaming for n seqs in group

Reply:
but currently imply that it's not supported